### ISSUE TITLE
Gamma Decoder with AMLP Recombination function fixes

### ISSUE THREAD

**Issue by sakishinoda** [sakishinoda/tf-ssl#1]
* [x] **Make Combinator class depend on u and z only**
Currently the u\*z multiplication for the augmented MLP recombination function is carried out in the Decoder whereas the Combinator should be a function g(u,z) and generate u\*z inside.

**Comment by sakishinoda** [comment]
**Adapt for remote running**

* [x] Make print statements flush. 
* [ ] Save intermediate checkpoints

**Comment by sakishinoda** [comment]
**Training terminating after 55000 steps**

* Supposed to be running for 100 epochs with batch size 100 (60k steps) then 50 more epochs (30k steps) with a linearly decaying learning rate 
* [ ] Train on validation set as well as training (to make up to 60k training examples)
* [x] Fix the decaying learning rate (perhaps using assign after 100 epochs)

**Comment by sakishinoda** [comment]
**CLI arguments**
* [ ] Specify model structure using string to tuple conversions
* [ ] EX_ID
* [ ] ???

**Comment by sakishinoda** [comment]
**Better training loop**

* [ ] Feed data using a queue? Serializing? TFRecords?
* [ ] Epoch -> step conversion
